src/function/scalar/string/regexp.cpp

```diff
@@ -203,6 +203,96 @@ static unique_ptr<FunctionData> RegexReplaceBind(ClientContext &context, ScalarF
 	return move(data);
 }
 
+static void RegexExtractFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto &func_expr = (BoundFunctionExpression &)state.expr;
+	auto &info = (RegexpExtractBindData &)*func_expr.bind_info;
+
+	auto &strings = args.data[0];
+	auto &patterns = args.data[1];
+	auto &groups = args.data[2];
+	if (info.constant_pattern) {
+		const RE2 re(duckdb_re2::StringPiece(info.pattern_string));
+		if (info.constant_group) {
+			const duckdb_re2::StringPiece rewrite(info.group_string);
+			UnaryExecutor::Execute<string_t, string_t>(strings, result, args.size(), [&](string_t input) {
+				std::string extracted;
```

Comment:
Can we extract these three lines to a function, since they are repeated 4X here?